### PR TITLE
8923 auto

### PR DESCRIPTION
**Pull request by sanjsaini**
We created, tested, and adjusted our Autonomous code. We have also set a new starting point.

### CODE REVIEW

File: Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/BlueCarouselDeliver.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        while (opModeIsActive()) {
+            double referenceAngle = imu.getAngularOrientation().firstAngle;
+            moveForward(-2.8, 10, 10);
+            spinCarouselBlue();
+            moveForward(30, 10, 10);
+            imuPivot(referenceAngle,-90, 35, 0.015, 3.0);
+
+            moveForward(5.0, 10, 10);
+            imuPivot(referenceAngle, 90, 35, 0.015, 3.0);
+            imuPivot(referenceAngle, 90, 35, 0.015, 3.0);
```
Comment (malavika1) [inline Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/BlueCarouselDeliver.java]: You shouldn't need this line twice

File: Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/BlueCarouselDeliverStorageUnit.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        while (opModeIsActive()) {
+            double referenceAngle = imu.getAngularOrientation().firstAngle;
+            moveForward(-2.8, 10, 10);
+            spinCarouselBlue();
+            moveForward(30, 10, 10);
+            imuPivot(referenceAngle,-90, 35, 0.015, 3.0);
+
+            moveForward(5.0, 10, 10);
+            imuPivot(referenceAngle, 90, 35, 0.015, 3.0);
+            imuPivot(referenceAngle, 90, 35, 0.015, 3.0);
```
Comment (malavika1) [inline Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/BlueCarouselDeliverStorageUnit.java]: I don't think you need this line twice

File: Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/BlueCarouselDeliverStorageUnit.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+            moveForward(-2.0, 10, 10);
+            autoDeliver();
+
+            //moving to the storage unit
```
Comment (malavika1) [inline Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/BlueCarouselDeliverStorageUnit.java]: What is this comment referring to? It looks like it refers to the break which doesn't make sense

File: Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/BlueStorageUnit.java [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+public class BlueStorageUnit extends MasterAutonomous {
+    public void runOpMode() {
+        while (opModeIsActive()) {
+            double referenceAngle = imu.getAngularOrientation().firstAngle;
+
+            moveForward(30, 10, 10);
+            imuPivot(referenceAngle, -90, 35, 0.015, 3.0);
+            moveForward(5, 10, 10);
+        }
+
```
Comment (sanjsaini) [inline Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/BlueStorageUnit.java]: Delete extra space.

File: Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/BlueCarouselDeliverWarehouse.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+        while (opModeIsActive()) {
+            double referenceAngle = imu.getAngularOrientation().firstAngle;
+            moveForward(-2.8, 10, 10);
+            spinCarouselBlue();
+            moveForward(30, 10, 10);
+            imuPivot(referenceAngle,-90, 35, 0.015, 3.0);
+
+            moveForward(5.0, 10, 10);
+            imuPivot(referenceAngle, 90, 35, 0.015, 3.0);
+            imuPivot(referenceAngle, 90, 35, 0.015, 3.0);
```
Comment (malavika1) [inline Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/BlueCarouselDeliverWarehouse.java]: You don't need this line twice

File: Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/BlueCarouselDeliverWarehouse.java [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+            moveForward(-2.0, 10, 10);
+            autoDeliver();
+            //move into warehouse
```
Comment (malavika1) [inline Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/BlueCarouselDeliverWarehouse.java]: This comment seems to refer to the break which doesn't make sense

File: Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/BlueDeliver.java [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+public class BlueDeliver extends MasterAutonomous {
+    public void runOpMode() throws InterruptedException {
+        initAuto();
+        waitForStart();
+
+        while (opModeIsActive()) {
+            double referenceAngle = imu.getAngularOrientation().firstAngle;
+            moveForward(5.0, 10, 10);
+            imuPivot(referenceAngle, 90, 35, 0.015, 3.0);
+            imuPivot(referenceAngle, 90, 35, 0.015, 3.0);
```
Comment (malavika1) [inline Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/BlueDeliver.java]: You don't need this line twice

File: Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/BlueStorageUnit.java (hunk 4 above)
Comment (sanjsaini) [reply Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/BlueStorageUnit.java]: More unnecessary spaces

File: Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/BlueDeliverStorageUnit.java [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+public class BlueDeliverStorageUnit extends MasterAutonomous {
+    public void runOpMode() throws InterruptedException {
+        initAuto();
+        waitForStart();
+
+        while (opModeIsActive()) {
+            double referenceAngle = imu.getAngularOrientation().firstAngle;
+            moveForward(5.0, 10, 10);
+            imuPivot(referenceAngle, 90, 35, 0.015, 3.0);
+            imuPivot(referenceAngle, 90, 35, 0.015, 3.0);
```
Comment (malavika1) [inline Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/BlueDeliverStorageUnit.java]: You don't need this line twice

File: Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/BlueDeliverWarehouse.java [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+public class BlueDeliverWarehouse extends MasterAutonomous {
+    public void runOpMode() throws InterruptedException {
+        initAuto();
+        waitForStart();
+
+        while (opModeIsActive()) {
+            double referenceAngle = imu.getAngularOrientation().firstAngle;
+            moveForward(5.0, 10, 10);
+            imuPivot(referenceAngle, 90, 35, 0.015, 3.0);
+            imuPivot(referenceAngle, 90, 35, 0.015, 3.0);
```
Comment (malavika1) [inline Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/BlueDeliverWarehouse.java]: You don't need this line twice

File: Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/MasterOpMode.java [hunk 10]
```diff
@@ -41,10 +44,17 @@ public void initHardware() {
         motorIntake = hardwareMap.dcMotor.get("motorIntake");
         motorIntake.setMode(DcMotor.RunMode.RUN_WITHOUT_ENCODER);
 
-
```
Comment (sanjsaini) [inline Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/MasterOpMode.java:44]: delete space

File: Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/MasterOpMode.java [hunk 11]
```diff
@@ -23,7 +27,6 @@ public void initHardware() {
         motorLeft = hardwareMap.dcMotor.get("motorLeft");
         motorRight = hardwareMap.dcMotor.get("motorRight");
 
-
```
Comment (sanjsaini) [inline Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/MasterOpMode.java:26]: delete space

File: Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/MasterOpMode.java [hunk 12]
```diff
@@ -4,16 +4,20 @@
...
 import com.qualcomm.robotcore.hardware.DcMotor;
 import com.qualcomm.robotcore.hardware.DcMotorSimple;
+import com.qualcomm.robotcore.hardware.Servo;
 
 abstract public class MasterOpMode extends LinearOpMode {
     //declare drive motors
     public DcMotor motorLeft = null;
     public DcMotor motorRight = null;
 
-
```
Comment (sanjsaini) [inline Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/MasterOpMode.java:13]: delete space

File: Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/BlueCarousel.java [hunk 13]
```diff
@@ -0,0 +1,17 @@
...
+public class BlueCarousel extends MasterAutonomous {
+    public void runOpMode() throws InterruptedException {
+        initAuto();
+        waitForStart();
+
+        while (opModeIsActive()) {
+            double referenceAngle = imu.getAngularOrientation().firstAngle;
+            moveForward(-2.8, 10, 10);
+            spinCarouselBlue();
+        }
```
Comment (malavika1) [inline Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/BlueCarousel.java]: You need a break at the end

File: Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/BlueDeliver.java [hunk 14]
```diff
@@ -0,0 +1,20 @@
...
+            moveForward(-2.0, 10, 10);
+            autoDeliver();
```
Comment (malavika1) [inline Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/BlueDeliver.java]: You need a break at the end

File: Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/BlueDeliverStorageUnit.java [hunk 15]
```diff
@@ -0,0 +1,22 @@
...
+            moveForward(-2.0, 10, 10);
+            autoDeliver();
+            imuPivot(referenceAngle, 90, 35, 0.015, 3.0);
+            moveForward(30, 10, 10);
+        }
```
Comment (malavika1) [inline Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/BlueDeliverStorageUnit.java]: You need a break at the end

File: Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/BlueDeliverWarehouse.java [hunk 16]
```diff
@@ -0,0 +1,21 @@
...
+            moveForward(-2.0, 10, 10);
+            autoDeliver();
+            imuPivot(referenceAngle, -90, 35, 0.015, 3.0);
```
Comment (malavika1) [inline Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/BlueDeliverWarehouse.java]: You need a break at the end

File: Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/BlueStorageUnit.java [hunk 17]
```diff
@@ -0,0 +1,14 @@
+package org.firstinspires.ftc.team8923_2021;
+
+public class BlueStorageUnit extends MasterAutonomous {
+    public void runOpMode() {
+        while (opModeIsActive()) {
+            double referenceAngle = imu.getAngularOrientation().firstAngle;
+
+            moveForward(30, 10, 10);
+            imuPivot(referenceAngle, -90, 35, 0.015, 3.0);
+            moveForward(5, 10, 10);
```
Comment (malavika1) [inline Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/BlueStorageUnit.java]: You need a break at the end

File: Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/BlueStorageUnit.java (hunk 4 above)
Comment (malavika1) [inline Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/BlueStorageUnit.java]: Delete extra line

File: Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/MasterTeleOp.java [hunk 18]
```diff
@@ -7,46 +7,68 @@
...
     }
 
     public void runDriveSpeed() {
         isSlowMode = driveSpeedToggle.toggle(gamepad1.left_bumper);
-        if (isSlowMode) driveSpeed = 0.25;
-        else driveSpeed = 1.0;
+        if (isSlowMode) {
+            driveSpeed = 0.25;
+        } else {
+            driveSpeed = 1.5;
```
Comment (malavika1) [inline Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/MasterTeleOp.java]: Motor power can't be more than 1.0

File: Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/MasterTeleOp.java [hunk 19]
```diff
@@ -7,46 +7,68 @@
...
+        if(gamepad2.left_bumper) {
+            motorLift.setPower(0.1);
+            servoGrabber.setPosition(1.0);
+        } else if (gamepad2.right_bumper){
+            motorLift.setPower(-0.1);
+            servoGrabber.setPosition(0.0);
+        } else {
+            motorLift.setPower(0.0);
+        }
+        /*if(gamepad2.left_bumper && !startScoring) {
```
Comment (malavika1) [inline Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/MasterTeleOp.java]: No commented out code

File: Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/RedCarousel.java [hunk 20]
```diff
@@ -0,0 +1,19 @@
...
+    public void runOpMode() throws InterruptedException {
+        initAuto();
+        waitForStart();
+
+        while (opModeIsActive()) {
+            double referenceAngle = imu.getAngularOrientation().firstAngle;
+
+            imuPivot(referenceAngle, 30, 35, 0.015, 3.0);
+            moveForward(-3.9, 10, 10);
+            spinCarouselRed();
```
Comment (malavika1) [inline Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/RedCarousel.java]: You need a break at the end

File: Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/RedDeliver.java [hunk 21]
```diff
@@ -0,0 +1,20 @@
...
+public class RedDeliver extends MasterAutonomous {
+    public void runOpMode() throws InterruptedException {
+        initAuto();
+        waitForStart();
+
+        while (opModeIsActive()) {
+            double referenceAngle = imu.getAngularOrientation().firstAngle;
+            moveForward(5.0, 10, 10);
+            imuPivot(referenceAngle, 90, 35, 0.015, 3.0);
+            imuPivot(referenceAngle, 90, 35, 0.015, 3.0);
```
Comment (malavika1) [inline Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/RedDeliver.java]: You don't need this line twice

File: Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/RedDeliver.java [hunk 22]
```diff
@@ -0,0 +1,20 @@
...
+            moveForward(-2.0, 10, 10);
+            autoDeliver();
```
Comment (malavika1) [inline Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/RedDeliver.java]: You need a break at the end

File: Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/RedStorageUnit.java [hunk 23]
```diff
@@ -0,0 +1,14 @@
+package org.firstinspires.ftc.team8923_2021;
+
+public class RedStorageUnit extends MasterAutonomous {
+    public void runOpMode() {
+        while (opModeIsActive()) {
+            double referenceAngle = imu.getAngularOrientation().firstAngle;
+
+            moveForward(30, 10, 10);
+            imuPivot(referenceAngle, 90, 35, 0.015, 3.0);
+            moveForward(5, 10, 10);
```
Comment (malavika1) [inline Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/RedStorageUnit.java]: You need a break at the end

File: Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/RedStorageUnit.java [hunk 24]
```diff
@@ -0,0 +1,14 @@
...
+        }
+
```
Comment (malavika1) [inline Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/RedStorageUnit.java]: Delete extra blank line

**Review by malavika1 — CHANGES_REQUESTED** [review]

File: Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/BlueCarousel.java [hunk 25]
```diff
@@ -0,0 +1,17 @@
...
+import com.qualcomm.robotcore.eventloop.opmode.Autonomous;
+
+@Autonomous(name="BlueCarousel")
+public class BlueCarousel extends MasterAutonomous {
+    public void runOpMode() throws InterruptedException {
+        initAuto();
+        waitForStart();
+
+        while (opModeIsActive()) {
+            double referenceAngle = imu.getAngularOrientation().firstAngle;
```
Comment (tmicro) [inline Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/BlueCarousel.java]: Note that the variable referenceAngle is not used in this method.

File: Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/TeleOpCompetition.java [hunk 26]
```diff
@@ -18,6 +17,7 @@ public void runOpMode()
             runDriveSpeed();
```
Comment (tmicro) [inline Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/TeleOpCompetition.java:17]: The runDriveSpeed() method sets the driveSpeed variable, but I can't see the variable being used anywhere in the class.

**Review by tmicro — COMMENTED** [review]
In general, autonomous programs should not have a while (opModeIsActive()) loop, since they are sequential, meaning that they execute a series of tasks once and then stop.  See the example FTC autonomous program in robotcontroller\external\samples\PushbotAutoDriveByEncoder_Linear.java

In contrast, TeleOp programs have the while (opModeIsActive()) loop because they are always checking for driver inputs and controlling the robot based on those inputs.

In the MasterAutonomous.java file, the initAuto() method hardware maps and initializes the imu.  This is redundant since the imu was already hardware mapped and initialized in MasterOpMode.java. initHardware() method.

In MasterAutonomous.java, the comment on line 177 states: loop until both motors are not busy.  However, the logic on line 178 does not implement that comment.  The logic && (AND) should change to || (OR).  See De Morgan's theorem in boolean algebra.

File: Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/BlueCarouselWarehouse.java [hunk 27]
```diff
@@ -0,0 +1,20 @@
...
+public class BlueCarouselWarehouse extends MasterAutonomous{
+    public void runOpMode() throws InterruptedException {
+        initAuto();
+        waitForStart();
+        double referenceAngle =  imu.getAngularOrientation().firstAngle;
+        moveForward(-2.8, 10, 10);
+        spinCarouselBlue();
+        moveForward(13, 50, 10);
+        imuPivot(referenceAngle, -90, 30, 0.015, 3.0);
+        moveForward(15, 50, 10); //check distance on this
```
Comment (malavika1) [inline Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/BlueCarouselWarehouse.java]: Have you checked the distance on this?

File: Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/BlueDeliver.java [hunk 28]
```diff
@@ -0,0 +1,17 @@
...
+
+@Autonomous(name="BlueDeliver")
+public class BlueDeliver extends MasterAutonomous {
+    public void runOpMode() throws InterruptedException {
+        initAuto();
+        waitForStart();
+        double referenceAngle = imu.getAngularOrientation().firstAngle;
+        moveForward(5.0, 10, 10);
+        imuPivot(referenceAngle, 90, 35, 0.015, 3.0);
+        imuPivot(referenceAngle, 90, 35, 0.015, 3.0);
```
Comment (malavika1) [inline Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/BlueDeliver.java]: This doesn't need to be repeated. It is doing the same thing twice.

File: Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/BlueDeliverStorageUnit.java [hunk 29]
```diff
@@ -0,0 +1,19 @@
...
+
+@Autonomous(name="BlueDeliverStorageUnit")
+public class BlueDeliverStorageUnit extends MasterAutonomous {
+    public void runOpMode() throws InterruptedException {
+        initAuto();
+        waitForStart();
+        double referenceAngle = imu.getAngularOrientation().firstAngle;
+        moveForward(-16.0, 10, 10);
+//      imuPivot(referenceAngle, , 35, 0.015, 3.0);
+        //moveForward(-2.0, 10, 10);
```
Comment (malavika1) [inline Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/BlueDeliverStorageUnit.java]: No commented out code

File: Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/BlueDeliverWarehouse.java [hunk 30]
```diff
@@ -0,0 +1,18 @@
...
+
+@Autonomous(name="BlueDeliverWarehouse")
+public class BlueDeliverWarehouse extends MasterAutonomous {
+    public void runOpMode() throws InterruptedException {
+        initAuto();
+        waitForStart();
+        double referenceAngle = imu.getAngularOrientation().firstAngle;
+        moveForward(-16.0, 10, 10);
+        /*imuPivot(referenceAngle, 180, 35, 0.015, 3.0);
+        moveForward(-2.0, 10, 10);*/
```
Comment (malavika1) [inline Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/BlueDeliverWarehouse.java]: No commented out code

File: Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/BlueWarehouse.java [hunk 31]
```diff
@@ -0,0 +1,16 @@
+package org.firstinspires.ftc.team8923_2021;
+
+import com.qualcomm.robotcore.eventloop.opmode.Autonomous;
+
+@Autonomous(name="BlueWarehouse")
+public class BlueWarehouse extends MasterAutonomous{
+    public void runOpMode() throws InterruptedException {
+        initAuto();
+        waitForStart();
+        //moveForward(16, 80, 10);
```
Comment (malavika1) [inline Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/BlueWarehouse.java]: No commented out code

File: Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/MasterAutonomous.java [hunk 32]
```diff
@@ -196,6 +198,33 @@ public void runIntake() {
...
+    }
+
+    public void spinCarouselRed() {
+        motorCarousel.setPower(-1.0);
+        sleep(4300);
+        motorCarousel.setPower(0.0);
+    }
+
+    public void autoDeliver() {
+            //motorLift.setTargetPosition(10);
```
Comment (malavika1) [inline Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/MasterAutonomous.java]: No commented out code

File: Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/MasterTeleOp.java [hunk 33]
```diff
@@ -1,52 +1,84 @@
...
 
-        } else motorCarousel.setPower(0.0);
+    public void grab() {
+        if (gamepad2.dpad_up) {
+            servoGrabber.setPosition(1.0);
+        } else if (gamepad2.dpad_down) {
+            servoGrabber.setPosition(0.0);
+        } else if (gamepad2.dpad_left) {
+            servoGrabber.setPosition(0.4);
+        }
```
Comment (malavika1) [inline Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/MasterTeleOp.java]: No blank line before }

File: Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/RedCarouselDeliverWarehouse.java [hunk 34]
```diff
@@ -0,0 +1,21 @@
...
+        initAuto();
+        waitForStart();
+        double referenceAngle = imu.getAngularOrientation().firstAngle;
+        imuPivot(referenceAngle, 30, 35, 0.015, 3.0);
+        moveForward(-3.9, 10, 10);
+        spinCarouselRed();
+        moveForward(35.0, 10, 10);
+        imuPivot(referenceAngle, 180, 35, 0.015, 3.0);
+        imuPivot(referenceAngle, -20, 35, 0.015, 3.0);
+        //moveForward(15, 10, 10);
```
Comment (malavika1) [inline Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/RedCarouselDeliverWarehouse.java]: No commented out code

File: Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/RedCarouselDeliverWarehouse.java [hunk 35]
```diff
@@ -0,0 +1,21 @@
...
+        autoDeliver();
+
```
Comment (malavika1) [inline Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/RedCarouselDeliverWarehouse.java]: No blank line before }

File: Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/RedCarouselWarehouse.java [hunk 36]
```diff
@@ -0,0 +1,18 @@
+package org.firstinspires.ftc.team8923_2021;
+
+import com.qualcomm.robotcore.eventloop.opmode.Autonomous;
+
+@Autonomous(name = "RedCarouselNoIntake")
```
Comment (malavika1) [inline Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/RedCarouselWarehouse.java]: This name doesn't match

File: Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/RedDeliverStorageUnit.java [hunk 37]
```diff
@@ -0,0 +1,19 @@
...
+
+@Autonomous(name="RedDeliverStorageUnit")
+public class RedDeliverStorageUnit extends MasterAutonomous {
+    public void runOpMode() throws InterruptedException {
+        initAuto();
+        waitForStart();
+        double referenceAngle = imu.getAngularOrientation().firstAngle;
+        moveForward(-16,10, 10);
+//      imuPivot(referenceAngle, , 35, 0.015, 3.0);
+        //moveForward(-2.0, 10, 10);
```
Comment (malavika1) [inline Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/RedDeliverStorageUnit.java]: No commented out code

File: Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/RedDeliverWarehouse.java [hunk 38]
```diff
@@ -0,0 +1,18 @@
...
+
+@Autonomous(name="RedDeliverWarehouse")
+public class RedDeliverWarehouse extends MasterAutonomous {
+    public void runOpMode() throws InterruptedException {
+        initAuto();
+        waitForStart();
+        double referenceAngle = imu.getAngularOrientation().firstAngle;
+        moveForward(-16, 10, 10);
+        /*imuPivot(referenceAngle, 180, 35, 0.015, 3.0);
+        moveForward(-2.0, 10, 10);*/
```
Comment (malavika1) [inline Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/RedDeliverWarehouse.java]: No commented out code

File: Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/RedStorageUnit.java [hunk 39]
```diff
@@ -0,0 +1,12 @@
+package org.firstinspires.ftc.team8923_2021;
+
+public class RedStorageUnit extends MasterAutonomous {
```
Comment (malavika1) [inline Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/RedStorageUnit.java]: This needs the @Autonomous tag

File: Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/RedWarehouse.java [hunk 40]
```diff
@@ -0,0 +1,15 @@
+package org.firstinspires.ftc.team8923_2021;
+
+import com.qualcomm.robotcore.eventloop.opmode.Autonomous;
+
+@Autonomous(name="RedWarehouseNoIntake")
```
Comment (malavika1) [inline Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/RedWarehouse.java]: This name doesn't match

**Review by malavika1 — CHANGES_REQUESTED** [review]

File: Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/BlueCarouselDeliverStorageUnit.java (hunk 3 above)
Comment (sanjsaini) [reply Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/BlueCarouselDeliverStorageUnit.java]: It was a reminder of what to do next in that auto code

File: Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/RedCarouselWarehouse.java [hunk 41]
```diff
@@ -0,0 +1,18 @@
...
+import com.qualcomm.robotcore.eventloop.opmode.Autonomous;
+
+@Autonomous(name = "RedCarouselWarehouse")
+public class RedCarouselWarehouse extends MasterAutonomous {
+    public void runOpMode() throws InterruptedException {
+        initAuto();
+        waitForStart();
+        double referenceAngle = imu.getAngularOrientation().firstAngle;
+        imuPivot(referenceAngle,30, 35, 0.015, 3.0);
+        moveForward(-3.9,10,10);
```
Comment (malavika1) [inline Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/RedCarouselWarehouse.java]: You should have a space after a comma

File: Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/RedDeliverStorageUnit.java [hunk 42]
```diff
@@ -0,0 +1,17 @@
...
+
+import com.qualcomm.robotcore.eventloop.opmode.Autonomous;
+
+@Autonomous(name="RedDeliverStorageUnit")
+public class RedDeliverStorageUnit extends MasterAutonomous {
+    public void runOpMode() throws InterruptedException {
+        initAuto();
+        waitForStart();
+        double referenceAngle = imu.getAngularOrientation().firstAngle;
+        moveForward(-16,10, 10);
```
Comment (malavika1) [inline Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/RedDeliverStorageUnit.java]: You should have a space after a comma

**Review by malavika1 — CHANGES_REQUESTED** [review]

File: Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/MasterTeleOp.java [hunk 43]
```diff
@@ -1,52 +1,84 @@
...
     private boolean isSlowMode = false;
 
     private Toggle driveSpeedToggle = new Toggle();
 
-    double driveSpeed = 1.0;
-
-    int scoreState = 0;
-    int flickStage = 0;
-
+    double driveSpeed = 1.5;
```
Comment (malavika1) [inline Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/MasterTeleOp.java]: Motor power can't be greater than 1.0

**Review by malavika1 — DISMISSED** [review]

File: Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/RedCarouselDeliverStorageUnit.java [hunk 44]
```diff
@@ -0,0 +1,22 @@
...
+        double referenceAngle = imu.getAngularOrientation().firstAngle;
+        imuPivot(referenceAngle, 30, 35, 0.015, 3.0);
+        moveForward(-3.9, 10, 10);
+        spinCarouselRed();
+        moveForward(30.0, 10, 10);
+        imuPivot(referenceAngle, 235, 35, 0.015, 3.0);
+        moveForward(-11, 10, 10);
+        autoDeliver();
+        moveForward(2, 10, 10);
+        imuPivot(referenceAngle,8,35, 0.015,3.0);
```
Comment (malavika1) [inline Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/RedCarouselDeliverStorageUnit.java]: Please add spaces after commas

File: Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/RedCarouselDeliverWarehouse.java [hunk 45]
```diff
@@ -0,0 +1,22 @@
...
+        double referenceAngle = imu.getAngularOrientation().firstAngle;
+        imuPivot(referenceAngle, 30, 35, 0.015, 3.0);
+        moveForward(-3.9, 10, 10);
+        spinCarouselRed();
+        moveForward(30.0, 10, 10);
+        imuPivot(referenceAngle, 235, 35, 0.015, 3.0);
+        moveForward(-11, 10, 10);
+        autoDeliver();
+        moveForward(2, 10, 10);
+        imuPivot(referenceAngle,8,35, 0.015,3.0);
```
Comment (malavika1) [inline Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/RedCarouselDeliverWarehouse.java]: Please add spaces after commas

**Review by malavika1 — APPROVED** [review]

**Pull request merged**